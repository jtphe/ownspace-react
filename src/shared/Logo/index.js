/* eslint-disable global-require */
import React from 'react';
import { StyleSheet, Image } from 'react-native';

/**
 * The Logo component
 */
const Logo = () => {
  /**
   * Render the Logo component
   * @returns {React.Component} - Logo component
   */
  return (
    <Image
      source={require('./images/os_logo.png')}
      style={styles.logo}
      resizeMode="contain"
    />
  );
};

/**
 * Styles of Logo component
 */
const styles = StyleSheet.create({
  logo: {
    alignSelf: 'center',
    marginLeft: -10,
    resizeMode: 'contain',
    width: 100,
    height: 100
  }
});
export default Logo;
