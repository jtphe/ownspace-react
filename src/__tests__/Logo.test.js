import 'react-native';
import React from 'react';
import Logo from '@shared/Logo';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Logo tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Logo />);
  });

  describe('Autocomplete snapshot', () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    renderer.create(<Logo />);
  });
});

Enzyme.configure({ adapter: new Adapter() });
