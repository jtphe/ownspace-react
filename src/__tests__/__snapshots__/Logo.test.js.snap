// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Image
  resizeMode="contain"
  source={1}
  style={
    Object {
      "alignSelf": "center",
      "height": 100,
      "marginLeft": -10,
      "resizeMode": "contain",
      "width": 100,
    }
  }
/>
`;
